indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,372 @@
+                                <span className="area-default">★</span>
+                            )}
+                        </>
+                    </Tooltip>
+                </Rectangle>
+            );
+        });
+    };
+
+    onSubmit = (data) => {
+        const {actions: {createMapArea, updateMapAreas}} = this.props;
+        const {areaProps: {id}} = this.state;
+        let result;
+
+        if (!id) {
+            result = createMapArea(data);
+        } else {
+            result = updateMapAreas([{...data, id}]);
+        }
+
+        result.then(() => {
+            this.setState({areaProps: null});
+        });
+    };
+
+    renderModalForm = () => {
+        const {areaProps} = this.state;
+        if (!areaProps) {
+            return null;
+        }
+
+        const {id, bounds, name, isDefault} = areaProps;
+        return (
+            <FinalForm onSubmit={this.onSubmit}
+                       render={this.renderForm}
+                       initialValues={{
+                           name: !id ? null : name,
+                           bounds,
+                           default: isDefault
+                       }} />
+        );
+    };
+
+    renderForm = (fprops) => {
+        const {
+            submitting, submitSucceeded, pristine, hasValidationErrors, handleSubmit
+        } = fprops;
+        const {areaProps: {id}} = this.state;
+        const isNewArea = !id;
+
+        return (
+            <Modal onClose={() => this.setState({areaProps: null})}
+                   closeIcon
+                   open>
+                <Modal.Header>
+                    {isNewArea ? (
+                        <Translate>Create new area</Translate>
+                    ) : (
+                        <Translate>Update area</Translate>
+                    )}
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="map-area-form" onSubmit={handleSubmit}>
+                        <Form.Group>
+                            <FinalInput name="name"
+                                        label={Translate.string('Name of the area')}
+                                        disabled={submitting || submitSucceeded}
+                                        autoFocus
+                                        required />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalCheckbox name="default"
+                                           label={Translate.string('Default map area')}
+                                           toggle />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalInput name="bounds.southWest.lat"
+                                        type="number"
+                                        label={Translate.string('SW Latitude')}
+                                        disabled={submitting || submitSucceeded}
```

Comment:
submitting is handled internally; no need to do it here

Reply:
Also, do we really want to disable on success? If we keep the modal open after saving (on edit), there's no need to prevent further edits. If we close it, it doesn't matter anyway.

Reply:
I close it so yeah, probably it is not that necessary here.